cameo/flux_analysis/simulation.py

```diff
@@ -75,10 +73,9 @@ def fba(model, objective=None, reactions=None, *args, **kwargs):
         Contains the result of the linear solver.
 
     """
-    with TimeMachine() as tm:
+    with model:
         if objective is not None:
-            tm(do=partial(setattr, model, 'objective', objective),
-               undo=partial(setattr, model, 'objective', model.objective))
+            model.objective = objective
         model.solver.optimize()
         if model.solver.status != 'optimal':
```

Comment:
I think it should raise the correct status: Infeasible, Unbounded, etc. We have some dictionaries inplace for this.
The type of exception is informative and necessary to decide how to handle the resolution.